### PR TITLE
fixed typos in R package docs

### PR DESCRIPTION
**Pull request by jameslamb**
Thank you for this great project. I hope you'll consider this pull request to fix minor typos in the R-package documentation (found with `devtools::spell_check()`).

I apologize about the extra lines touched for trailing whitespace, that is a setting I have in my IDE. Please let me know if you would like me to revert those whitespace-only edits.

I also regenerated the documentation with `devtools:;document()` so the `.Rd` files in `man/` would be consistent with the roxygen changes I made.

### CODE REVIEW

**Comment by jameslamb** [comment]
I see that my build is failing with this error:

image

This is an error I saw a lot in my own projects (private and open source) when [devtools went to 2.0.0](https://github.com/r-lib/devtools/releases/tag/v2.0.0).

This is a result of the use of `craigcitro/r-travis in` this project's CI. That project [has been deprecated](https://github.com/craigcitro/r-travis/commit/dd538f85b2d3ca42e6587beb363f2ea54de5bb55) since 2016.

Do you have an issue open to replace it? I would love to contribute a fix!

**Comment by hcho3** [comment]
@jameslamb That would be great!

**Comment by trivialfis** [comment]
@jameslamb Thanks, please go ahead and submit a PR.

**Comment by jameslamb** [comment]
I've rebased this to `master` now that #4353 has been merged. Should build successfully now.

**Comment by jameslamb** [comment]
I think Travis is failing because of that same transient failure in  the `python_test` stage I saw on #4353 and others.

File: R-package/man/xgb.train.Rd [hunk 1]
```diff
@@ -210,7 +211,7 @@ dtest <- xgb.DMatrix(agaricus.test$data, label = agaricus.test$label)
 watchlist <- list(train = dtrain, eval = dtest)
 
 ## A simple xgb.train example:
-param <- list(max_depth = 2, eta = 1, verbosity = 0, nthread = 2, 
+param <- list(max_depth = 2, eta = 1, silent = 1, nthread = 2,
```
Comment (trivialfis) [inline R-package/man/xgb.train.Rd]: The silent parameter is deprecated, see https://xgboost.readthedocs.io/en/latest/parameter.html .

Comment (jameslamb) [reply R-package/man/xgb.train.Rd]: @trivialfis oh this is interesting. I didn't actually touch this line directly. File changes in `.Rd` files were auto-generated when I ran `devtools::document()`.

Maybe I should rebase and try again?

Comment (jameslamb) [reply R-package/man/xgb.train.Rd]: ohhhhh I wonder if someone in a previous PR manually edited the `.Rd` but not the corresponding roxygen block in the R code? I'll fix that on this PR

Comment (jameslamb) [reply R-package/man/xgb.train.Rd]: Ok yes, this is what happened! I walked through the blame and found that this parameter was changed in `xgb.train.Rd` in #4267 but not the corresponding roxygen content in the R source code.

Comment (jameslamb) [reply R-package/man/xgb.train.Rd]: fixed in https://github.com/dmlc/xgboost/pull/4345/commits/25dbe3a9ccd6ed0d5d08c5a9c459a46d33b2ec1b

Comment (trivialfis) [reply R-package/man/xgb.train.Rd]: @jameslamb Thanks for fixing my mistake!  I restarted the test and will merge once they are done.

Comment (jameslamb) [reply R-package/man/xgb.train.Rd]: No problem, happy to help!

**Comment by trivialfis** [comment]
Thanks!

**Pull request merged**